src/main/kotlin/me/partlysanestudios/partlysaneskies/features/discord/webhooks/WebhookMenu.kt

```diff
@@ -0,0 +1,356 @@
+            webhookIcons.add(icon)
+        }
+    }
+
+    private fun updateSelected() {
+        updateParameters()
+
+        for (icon in webhookIcons) {
+            icon.toggle.setState(false)
+        }
+
+        selectedIcon?.toggle?.setState(true)
+
+        if (selectedIcon != null) {
+            webhookOptionsHeader.setText("§e${selectedIcon?.webhookEvent?.name ?: ""} Options:")
+        }
+    }
+
+    private fun updateParameters() {
+        val selectedConfig = selectedIcon?.webhookEvent?.config
+        if (selectedConfig == null) {
+            for (param in activeParameters) {
+                param.parameterBlock.hide(true)
+                param.textComponent.hide(true)
+                param.toggle.hide(true)
+                webhookOptionsHeader.setText("")
+            }
+            activeParameters.clear()
+        } else {
+            val startX = 3.0
+
+            var x = startX
+            var y = 30
+            val width = 20
+            for (key in selectedConfig.getAllOptions().keys) {
+
+                if (x + width > 100) {
+                    y += 30
+                    x = startX
+                }
+
+                if (selectedConfig.getAllOptions()[key] !is Toggle) {
+                    continue
+                }
+
+                val option = WebhookOption(selectedConfig, key)
+                option.menu = this
+                option.parameterBlock.constrain {
+                    this.x = x.percent
+                    this.y = y.scaledPixels
+                    this.width = width.percent
+                    height = 20.scaledPixels
+                    color = Color(0, 0, 0, 0).constraint
+                }.setChildOf(webhookOptionsList)
+
+                activeParameters.add(option)
+
+
+                x += 85.0 / 3.0 + 3.1
+            }
+        }
+    }
+    private fun updateLocations() {
+        val enabled = ArrayList<WebhookIcon>()
+        val disabled = ArrayList<WebhookIcon>()
+        for (icon in webhookIcons) {
+            if (icon.webhookEvent.hidden && !config.showHiddenWebhooks) {
+                continue
+            }
+
+            if (icon.enabled) {
+                enabled.add(icon)
+            } else {
+                disabled.add(icon)
+            }
+        }
+
+        val startX = 10.0
+
+        var x = startX
+        var y = 30.0
+        val width = 30
+        val height = 30
+        val pad = 7.5
+
+        for (icon in disabled) {
+            icon.iconBox
+                .setX(x.scaledPixels)
+                .setY(y.scaledPixels)
+                .setWidth(width.scaledPixels)
+                .setHeight(height.scaledPixels)
+                .setChildOf(inactiveWebhooksPanel)
+
+            x += width + pad
+            if (x + width > 300) {
+                x = startX
+                y += height + pad
+            }
+        }
+
+        x = startX
+        y = 25.0
+        for (icon in enabled) {
+            icon.iconBox
```

Comment:
are you able to use .constraint{} on this

Reply:
No

Reply:
Okay